internal/cel/compiler_test.go

```diff
@@ -192,3 +204,321 @@ func TestCompiledProgram_GetExpression(t *testing.T) {
+			expectError: false,
+		},
+		{
+			name:        "valid annotation value - multiline",
+			expression:  `annotation("valid-key", "line1\nline2\nline3")`,
+			expectError: false,
+		},
+		// Note: Large annotation value testing is done in TestValidationFunctions
+		// to avoid CEL expression size limits
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			// Compile the expression
+			ast, issues := env.Compile(tt.expression)
+			g.Expect(issues.Err()).NotTo(HaveOccurred(), "Expression should compile successfully")
+
+			// Create program and evaluate
+			program, err := env.Program(ast)
+			g.Expect(err).NotTo(HaveOccurred(), "Program creation should succeed")
+
+			// Evaluate the expression
+			result, _, err := program.Eval(map[string]interface{}{})
+
+			if tt.expectError {
+				g.Expect(err).To(HaveOccurred(), "Expected evaluation to fail")
+				g.Expect(err.Error()).To(ContainSubstring(tt.errorMsg))
+			} else {
+				g.Expect(err).NotTo(HaveOccurred(), "Expected evaluation to succeed")
+				g.Expect(result).NotTo(BeNil(), "Expected valid result")
+			}
+		})
+	}
+}
+
+func TestValidationFunctions(t *testing.T) {
+	g := NewWithT(t)
+
+	// Test validateLabelValue
+	t.Run("validateLabelValue", func(t *testing.T) {
+		// Valid label values
+		g.Expect(validateLabelValue("")).To(Succeed())
+		g.Expect(validateLabelValue("valid123")).To(Succeed())
+		g.Expect(validateLabelValue("valid-value_with.dots")).To(Succeed())
+
+		// Invalid label values
+		err := validateLabelValue("-invalid")
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(err.Error()).To(ContainSubstring("label value"))
+
+		err = validateLabelValue("invalid-")
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(err.Error()).To(ContainSubstring("label value"))
+
+		err = validateLabelValue(strings.Repeat("a", maxLabelValueLength+1))
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(err.Error()).To(ContainSubstring("label value"))
+
+		err = validateLabelValue("invalid/value")
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(err.Error()).To(ContainSubstring("label value"))
+	})
+
+	// Test validateAnnotationValue
+	t.Run("validateAnnotationValue", func(t *testing.T) {
+		// Valid annotation values
+		g.Expect(validateAnnotationValue("")).To(Succeed())
+		g.Expect(validateAnnotationValue("valid value with spaces")).To(Succeed())
+		g.Expect(validateAnnotationValue("测试中文字符 🚀")).To(Succeed())
+		g.Expect(validateAnnotationValue("line1\nline2\nline3")).To(Succeed())
+		g.Expect(validateAnnotationValue(strings.Repeat("a", maxAnnotationValueSize))).To(Succeed())
+
+		// Invalid annotation value - too long
+		err := validateAnnotationValue(strings.Repeat("a", maxAnnotationValueSize+1))
+		g.Expect(err).To(HaveOccurred())
+		g.Expect(err.Error()).To(ContainSubstring("annotation value is too long"))
+		g.Expect(err.Error()).To(ContainSubstring("262145 bytes"))
+		g.Expect(err.Error()).To(ContainSubstring("maximum allowed is 262144 bytes"))
```

Comment:
alternative notation

```suggestion
		g.Expect(validateAnnotationValue(strings.Repeat("a", maxAnnotationValueSize+1))).
			Error().To(MatchError(And(
			ContainSubstring("annotation value is too long"),
			ContainSubstring("262145 bytes"),
			ContainSubstring("maximum allowed is 262144 bytes"))))
```